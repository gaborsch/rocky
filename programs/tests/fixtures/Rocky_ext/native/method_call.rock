from java, math play BigDecimal

X will be BigDecimal taking "1.0"
Y will be BigDecimal taking 2
say "Instantiation OK"

Let Z be add on X taking Y
say "direct method call OK"

Let Z1 be toString on Z
Say Z1
say "parameterless method call OK"

Let Z2 be shortValue on Z
Say Z2
say "inherited method call OK"

Let Z3 be valueOf on BigDecimal taking 10, 3
Say toString on Z3
Let Z3 be valueOf on BigDecimal taking 10, 0
Say toString on Z3
Let Z31 be valueOf on BigDecimal taking 10, -3
Say toString on Z31
say "static method OK"

Let Z4 be valueOf on Z taking 10, 2
Say toString on Z4
say "static method on instance OK"

Let Z5 be ROUND_DOWN on Z
Say Z5
say "property access OK"

Let Z6 be tostring on Z
Say Z6
say "case-insensitive method call OK"

Let Z7 be Short Value on Z
Say Z7
say "method call with spaces OK"

let Z8 be get class on Z
Say Z8
say "getter call with spaces OK"

