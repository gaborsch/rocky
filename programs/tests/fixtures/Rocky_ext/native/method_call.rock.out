Instantiation OK
direct method call OK
3.0
parameterless method call OK
3
inherited method call OK
0.010
10
1.0E+4
static method OK
0.10
static method on instance OK
1
property access OK
3.0
case-insensitive method call OK
3
method call with spaces OK
1
getter call with spaces OK
